tasks/configure.yml

```diff
@@ -122,6 +122,78 @@
   register: postgresql_configuration_pt3
   changed_when: "'_OK_' not in postgresql_configuration_pt3.stdout"
 
+- name: PostgreSQL | Generate SSL Private Key | RedHat >= v7
+  openssl_privatekey:
+    path: "{{ postgresql_ssl_key_file }}"
+    type: "RSA"
+    size: "{{ postgresql_ssl_key_length }}"
+  become: true
+  become_user: "{{ postgresql_service_user }}"
+  when: ansible_os_family == "RedHat" and
+        ansible_distribution_major_version | version_compare('7','>=') and
+        postgresql_ssl
+
+- name: PostgreSQL | Generate SSL Private Key | RedHat <= v6
```

Comment:
It's not my favourite option, but it seems that getting pyopenssl working on CentOS v6 is problematic.  This falls back to my original solution, which was to use `shell:` and `openssl`.